### PR TITLE
feat: update examples for epoch 3.0

### PR DESCRIPTION
**Pull request by leahjlou**

### CODE REVIEW

File: examples/nft-marketplace/contracts/nft-marketplace.clar [hunk 1]
```diff
@@ -96,7 +96,7 @@
     ;; Verify that the contract of this asset is whitelisted
     (asserts! (is-whitelisted (contract-of nft-asset-contract)) ERR_ASSET_CONTRACT_NOT_WHITELISTED)
     ;; Verify that the asset is not expired
-    (asserts! (> (get expiry nft-asset) block-height) ERR_EXPIRY_IN_PAST)
+    (asserts! (> (get expiry nft-asset) tenure-height) ERR_EXPIRY_IN_PAST)
```
Comment (hugocaillard) [inline examples/nft-marketplace/contracts/nft-marketplace.clar:99]:
```suggestion
    (asserts! (> (get expiry nft-asset) burn-block-height) ERR_EXPIRY_IN_PAST)
```

File: examples/nft-marketplace/contracts/nft-marketplace.clar [hunk 2]
```diff
@@ -174,7 +174,7 @@
       ERR_UNINTENDED_TAKER
     )
     ;; Verify the listing for purchase is not expired
-    (asserts! (< block-height (get expiry listing)) ERR_LISTING_EXPIRED)
+    (asserts! (< tenure-height (get expiry listing)) ERR_LISTING_EXPIRED)
```
Comment (hugocaillard) [inline examples/nft-marketplace/contracts/nft-marketplace.clar:177]:
```suggestion
    (asserts! (< burn-block-height (get expiry listing)) ERR_LISTING_EXPIRED)
```

File: examples/stx-defi/contracts/stx-defi.clar [hunk 3]
```diff
@@ -138,7 +138,7 @@
 (define-private (calculate-accrued-interest (principal uint) (start-block uint))
     (let (
         ;; Calculate the number of blocks elapsed since the last interaction.
-        (elapsed-blocks (- block-height start-block))
+        (elapsed-blocks (- tenure-height start-block))
```
Comment (hugocaillard) [inline examples/stx-defi/contracts/stx-defi.clar:141]:
```suggestion
        (elapsed-blocks (- burn-block-height start-block))
```

File: examples/stx-defi/contracts/stx-defi.clar [hunk 4]
```diff
@@ -107,7 +107,7 @@
         ;; Transfer the repayment amount from the user to the contract.
         (try! (stx-transfer? amount tx-sender (as-contract tx-sender)))
         ;; Update the user's loan details in the map with the new total due.
-        (map-set loans tx-sender { amount: (- total-due amount), last-interaction-block: block-height })
+        (map-set loans tx-sender { amount: (- total-due amount), last-interaction-block: tenure-height })
```
Comment (hugocaillard) [inline examples/stx-defi/contracts/stx-defi.clar:110]:
```suggestion
        (map-set loans tx-sender { amount: (- total-due amount), last-interaction-block: burn-block-height })
```

File: examples/stx-defi/contracts/stx-defi.clar [hunk 5]
```diff
@@ -61,7 +61,7 @@
             (try! (as-contract (stx-transfer? amount tx-sender recipient)))
         )
         ;; Update the user's loan details in the map.
-        (map-set loans tx-sender { amount: new-loan, last-interaction-block: block-height })
+        (map-set loans tx-sender { amount: new-loan, last-interaction-block: tenure-height })
```
Comment (hugocaillard) [inline examples/stx-defi/contracts/stx-defi.clar:64]:
```suggestion
        (map-set loans tx-sender { amount: new-loan, last-interaction-block: burn-block-height })
```

**Review by hugocaillard — CHANGES_REQUESTED** [review]
Since the contracts using `block-height` were mostly using it to mesure time, in think `burn-block-height` makes mored sense. And could have been use since the beginning, because stacks blocks where never meant to be mined at a regular pace (except that it was following bitcoin blocks)

File: examples/semi-fungible-token/deployments/default.simnet-plan.yaml [hunk 6]
```diff
@@ -65,85 +65,85 @@ plan:
       epoch: "2.1"
     - id: 2
       transactions: []
-      epoch: "2.1"
+      epoch: "3.0"
```
Comment (hugocaillard) [inline examples/semi-fungible-token/deployments/default.simnet-plan.yaml:68]: All of the empty batches in this files are a clarinet bug, I'll clean manually untils fixed in clarinet

**Comment by hugocaillard** [comment]
Added a commit that updated the npm dependencies and cleaned some of the examples

**Comment by leahjlou** [comment]
Thanks @hugocaillard, I just updated these to use `burn-block-height`. I agree that it makes more sense here.

**Review by hugocaillard — APPROVED** [review]
LGTM

**Pull request merged**